sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -227,34 +253,74 @@ void gnrc_rpl_send_DIO(gnrc_rpl_instance_t *inst, ipv6_addr_t *destination)
     gnrc_rpl_send(pkt, dodag->iface, NULL, destination, &dodag->dodag_id);
 }
 
-void gnrc_rpl_send_DIS(gnrc_rpl_instance_t *inst, ipv6_addr_t *destination)
+void gnrc_rpl_send_DIS(gnrc_rpl_instance_t *inst, ipv6_addr_t *destination,
+                       gnrc_rpl_internal_opt_t **options, size_t num_opts)
 {
-    gnrc_pktsnip_t *pkt;
+    gnrc_pktsnip_t *pkt = NULL, *tmp;
     icmpv6_hdr_t *icmp;
     gnrc_rpl_dis_t *dis;
 
-    /* TODO: Currently the DIS is too small so that wireshark complains about an incorrect
-     * ethernet frame check sequence. In order to prevent this, 4 PAD1 options are added.
-     * This will be addressed in follow-up PRs */
-    uint8_t padding[] = {
-            0x01, 0x02, 0x00, 0x00
-    };
-
-    int size = sizeof(icmpv6_hdr_t) + sizeof(gnrc_rpl_dis_t) + sizeof(padding);
+    /* No options provided to be attached to the DIS, so we PadN 2 bytes */
+    if (options == NULL || num_opts == 0) {
+        assert(!options);
+        gnrc_pktsnip_t *opt_snip;
+        size_t snip_size = 0;
+        /* The DIS is too small so that wireshark complains about an incorrect
+         * ethernet frame check sequence.
+         * To trick it we PadN 2 additional bytes, i.e. 4 bytes in sum. */
+        uint8_t padding[] = {
+            GNRC_RPL_OPT_PADN,  /* Option Type */
+            0x02,               /* Number of extra padding bytes */
+            0x00, 0x00
+        };
+
+        snip_size = sizeof(padding);
+        if ((opt_snip = gnrc_pktbuf_add(pkt, NULL, snip_size,
+                                    GNRC_NETTYPE_UNDEF)) == NULL) {
+            DEBUG("RPL: BUILD PadN OPT - no space left in packet buffer\n");
+            gnrc_pktbuf_release(pkt);
+            return;
+        }
+        memcpy(opt_snip->data, padding, snip_size);
+        pkt = opt_snip;
+    }
+    else {
+        assert(options);
+        for (size_t i = 0; i < num_opts; ++i) {
+            switch(options[i]->type) {
```

Comment:
`switch case` is rather expensive compared to `if else`, especially as currently only one case is checked. I would go for the `if` variant (until more cases exists), IMO this is simpler and more readable (for now).